File: app/src/main/java/com/derek_s/hubble_gallery/model/Tiles.java
package com.derek_s.hubble_gallery.model;

import com.google.gson.Gson;

import java.util.ArrayList;

/**
 * Created by dereksmith on 15-03-03.
 */
public class Tiles extends SerializableModel{

    private ArrayList<TileObject> tiles = new ArrayList<>();

    /**
     * @return The tiles
     */
    public ArrayList<TileObject> getTiles() {
        return tiles;
    }

    /**
     * @param tiles The chats
     */
    public void setTiles(ArrayList<TileObject> tiles) {
        this.tiles = tiles;
    }

    public String serialize() {
        // Serialize this class into a JSON string using GSON
        super();
    }

    static public DetailsObject create(String serializedData) {
        // Use GSON to instantiate this class using the JSON representation of the state
            super(serializedData);
    }
}

File: app/src/main/kotlin/com/derek_s/hubble_gallery/model/Tiles.kt
package com.derek_s.hubble_gallery.model

import com.google.gson.Gson
import java.util.*

class Tiles {

  var tiles: ArrayList<TileObject> = ArrayList()

  fun serialize(): String {
    val gson = Gson()
    return gson.toJson(this)
  }

  companion object {
    fun create(serializedData: String): Tiles {
      val gson = Gson()
      return gson.fromJson(serializedData, Tiles::class.java)
    }
  }
}